From 9c74bae21a7af11a0b0de40d87e654f6dbb39707 Mon Sep 17 00:00:00 2001
From: Robert Marko <robert.marko@sartura.hr>
Date: Wed, 20 Oct 2021 14:29:37 +0200
Subject: [PATCH] arm: dts: ipq4019: add switch nodes

Since the built-in IPQ40xx switch now has a driver, add the
required nodes for it to work.

Signed-off-by: Robert Marko <robert.marko@sartura.hr>
---
 arch/arm/boot/dts/qcom-ipq4019.dtsi | 87 +++++++++++++++++++++++++++++
 1 file changed, 87 insertions(+)

--- a/arch/arm/boot/dts/qcom-ipq4019.dtsi
+++ b/arch/arm/boot/dts/qcom-ipq4019.dtsi
@@ -589,6 +589,84 @@
 			status = "disabled";
 		};
 
+		switch: switch@c000000 {
+			compatible = "qca,qca8337-mmio";
+			reg = <0xc000000 0x80000>;
+			resets = <&gcc ESS_RESET>;
+			reset-names = "ess_rst";
+			clocks = <&gcc GCC_ESS_CLK>;
+			clock-names = "ess_clk";
+			psgmii-phy = <&psgmii_phy>;
+			mdio = <&mdio>;
+			mac-mode = <0>; /* 0 = PSGMII, 1 = RGMII5 */
+
+			status = "disabled";
+
+			ports {
+				#address-cells = <1>;
+				#size-cells = <0>;
+
+				port@0 { /* MAC0 */
+					reg = <0>;
+					label = "cpu";
+					ethernet = <&gmac>;
+					phy-mode = "internal";
+
+					fixed-link {
+						speed = <1000>;
+						full-duplex;
+						pause;
+						asym-pause;
+					};
+				};
+
+				swport1: port@1 { /* MAC1 */
+					reg = <1>;
+					label = "lan1";
+					phy-handle = <&ethphy0>;
+					phy-mode = "psgmii";
+
+					status = "disabled";
+				};
+
+				swport2: port@2 { /* MAC2 */
+					reg = <2>;
+					label = "lan2";
+					phy-handle = <&ethphy1>;
+					phy-mode = "psgmii";
+
+					status = "disabled";
+				};
+
+				swport3: port@3 { /* MAC3 */
+					reg = <3>;
+					label = "lan3";
+					phy-handle = <&ethphy2>;
+					phy-mode = "psgmii";
+
+					status = "disabled";
+				};
+
+				swport4: port@4 { /* MAC4 */
+					reg = <4>;
+					label = "lan4";
+					phy-handle = <&ethphy3>;
+					phy-mode = "psgmii";
+
+					status = "disabled";
+				};
+
+				swport5: port@5 { /* MAC5 */
+					reg = <5>;
+					label = "wan";
+					phy-handle = <&ethphy4>;
+					phy-mode = "psgmii";
+
+					status = "disabled";
+				};
+			};
+		};
+
 		gmac: ethernet@c080000 {
 			compatible = "qcom,ipq4019-ess-edma";
 			reg = <0xc080000 0x8000>;
@@ -664,6 +742,15 @@
 			};
 		};
 
+		psgmii_phy: psgmii-phy@98000 {
+			compatible = "qcom,ipq4019-psgmii-phy", "syscon";
+			reg = <0x98000 0x800>;
+			resets = <&gcc ESS_PSGMII_ARES>;
+			reset-names = "psgmii_rst";
+
+			status = "disabled";
+		};
+
 		usb3_ss_phy: ssphy@9a000 {
 			compatible = "qcom,usb-ss-ipq4019-phy";
 			#phy-cells = <0>;
